Warnings when building

```
g++ -std=c++11 -Wall -O2 -flto -DVERSION=v0.12-73-gcefe44a -Iutf8proc -Igopt   -std=c++11 -pthread -I/usr/local/Cellar/re2/20180801/include  -o src/string.o -c src/string.cpp
src/status.cpp:33:18: warning: ISO C++11 does not allow conversion from string literal to 'char *' [-Wwritable-strings]
  out = tigetstr("setaf");                                                                                              
                 ^                               
src/status.cpp:169:21: warning: ISO C++11 does not allow conversion from string literal to 'char *' [-Wwritable-strings]
    cuu1 = tigetstr("cuu1"); // cursor up one row                                                                       
                    ^                                        
src/status.cpp:170:19: warning: ISO C++11 does not allow conversion from string literal to 'char *' [-Wwritable-strings]
    cr = tigetstr("cr");     // return to first column                                                                  
                  ^                                       
src/status.cpp:171:19: warning: ISO C++11 does not allow conversion from string literal to 'char *' [-Wwritable-strings]
    ed = tigetstr("ed");     // erase to bottom of display                                                              
                  ^                                        
src/status.cpp:172:21: warning: ISO C++11 does not allow conversion from string literal to 'char *' [-Wwritable-strings]
    rows = tigetnum("lines");                                                                                                             
                    ^                                                                                                                                   
src/status.cpp:173:21: warning: ISO C++11 does not allow conversion from string literal to 'char *' [-Wwritable-strings]                                       
    cols = tigetnum("cols");                                                                                                       
                    ^                                                                                                                                                                                                          
src/status.cpp:178:21: warning: ISO C++11 does not allow conversion from string literal to 'char *' [-Wwritable-strings]
    sgr0 = tigetstr("sgr0"); // optional                                                                                                                                    
                    ^
```

--- a/src/location.cpp
+++ b/src/location.cpp
@@ -17,12 +17,13 @@ std::ostream & operator << (std::ostream &os, TextLocation location) {
   const Location *l = location.l;
   FILE *f;
   char buf[40];
-  long get;
+  size_t get;
 
   if (l->start.bytes >= 0 &&
       l->filename[0] != '<' &&
       l->start.row == l->end.row &&
-      0 < (get = l->end.column - l->start.column + 1) && get < sizeof(buf) &&
+      l->end.column >= l->start.column &&
+      (get = l->end.column - l->start.column + 1) < sizeof(buf) &&
       (f = fopen(l->filename, "r")) != 0) {
     if (fseek(f, l->start.bytes, SEEK_SET) != -1 && fread(&buf[0], 1, get, f) == get) {
       buf[get] = 0;

--- a/src/sources.cpp
+++ b/src/sources.cpp
@@ -51,7 +51,10 @@ static std::string slurp(int dirfd, bool &fail) {
     fail = true;
     fprintf(stderr, "Failed to open pipe %s\n", strerror(errno));
   } else if ((pid = fork()) == 0) {
-    fchdir(dirfd);
+    if (fchdir(dirfd) == -1) {
+      fprintf(stderr, "Failed to chdir: %s\n", strerror(errno));
+      exit(1);
+    }
     close(pipefd[0]);
     if (pipefd[1] != 1) {
       dup2(pipefd[1], 1);

--- a/src/status.cpp
+++ b/src/status.cpp
@@ -28,9 +28,10 @@ static int used = 0;
 
 const char *term_red()
 {
+  static char setaf_lit[] = "setaf";
   char *out;
   if (!sgr0) return "";
-  out = tigetstr("setaf");
+  out = tigetstr(setaf_lit);
   if (!out || out == (char*)-1) return "";
   out = tparm(out, 1); // red
   if (!out || out == (char*)-1) return "";
@@ -166,16 +167,23 @@ void term_init(bool tty_)
   }
 
   if (tty) {
-    cuu1 = tigetstr("cuu1"); // cursor up one row
-    cr = tigetstr("cr");     // return to first column
-    ed = tigetstr("ed");     // erase to bottom of display
-    rows = tigetnum("lines");
-    cols = tigetnum("cols");
+    // tigetstr function argument is (char*) on some platforms, so we need this hack:
+    static char cuu1_lit[] = "cuu1";
+    static char cr_lit[] = "cr";
+    static char ed_lit[] = "ed";
+    static char lines_lit[] = "lines";
+    static char cols_lit[] = "cols";
+    static char sgr0_lit[] = "sgr0";
+    cuu1 = tigetstr(cuu1_lit);   // cursor up one row
+    cr   = tigetstr(cr_lit);     // return to first column
+    ed   = tigetstr(ed_lit);     // erase to bottom of display
+    rows = tigetnum(lines_lit);
+    cols = tigetnum(cols_lit);
     if (!cuu1 || cuu1 == (char*)-1) tty = false;
     if (!cr || cr == (char*)-1) tty = false;
     if (!ed || ed == (char*)-1) tty = false;
     if (cols < 0 || rows < 0) tty = false;
-    sgr0 = tigetstr("sgr0"); // optional
+    sgr0 = tigetstr(sgr0_lit); // optional
     if (sgr0 == (char*)-1) sgr0 = 0;
   }
 }

